@xen-orchestra/web-core/lib/components/fullscreenable/UiFullscreenable.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <Teleport to="body" :disabled="!isMobile">
+    <div v-if="isOpen && isMobile" class="mobile">
+      <FullscreenableHeader v-if="isMobile" :back="close"><slot name="header" /></FullscreenableHeader>
+      <slot />
+    </div>
+    <slot v-else />
+  </Teleport>
+</template>
+
+<script lang="ts" setup>
+import FullscreenableHeader from '@core/components/fullscreenable/FullscreenableHeader.vue'
+import { ref, inject } from 'vue'
+import { useUiStore } from '@core/stores/ui.store'
+import { storeToRefs } from 'pinia'
+import { IK_FULLSCREENABLE_BEFORE_CLOSE } from '@core/utils/injection-keys.util'
+import { noop } from '@vueuse/core'
+
+const { isMobile } = storeToRefs(useUiStore())
+
+const beforeClose = inject(IK_FULLSCREENABLE_BEFORE_CLOSE, noop)
+
+const isOpen = ref(true)
+const close = () => {
+  beforeClose()
+  isOpen.value = false
+}
+</script>
+
+<style lang="postcss" scoped>
+.mobile {
+  background-color: var(--background-color-secondary);
+  height: 100vh;
+  position: absolute;
+  top: 0;
+  width: 100%;
+  z-index: 10001;
+}
+</style>
```

Comment:
I think that this component should not be responsible of its state and thus, not to contain `isOpen`/`close`.

IMO, it should only be a wrapper to display an element in fullscreen on mobile, with a `disabled` prop to handle case where we don't want to teleport the content.

Maybe it should be named accordingly to that statement (for example `MobileFullscreen`. **In my next comments, I will use this name for clarity**, but it can be changed if we think about a better name)

Here is some suggestions (it's just a mock, there is missing imports, non-existant components etc.), the comments are here for explanation and can be removed:

```suggestion
<template>
  <!-- When disabled or viewed on desktop, display the slot as is -->
  <slot v-if="uiStore.isDesktop || disabled">
  <!-- In other cases, if this "panel" is `open`, let's display it in the body -->
  <Teleport v-if="open" to="body">
    <div class="mobile-fullscreen">
      <!-- MobileHeadBar is a designed component in Figma -->
      <!-- It's just used here as example and doesn't have to be created now -->
      <MobileHeadBar>
        <ButtonIcon :icon="faAngleLeft" @click="emit('close')" />
        <slot name="header" />
      </MobileHeadBar>
      <slot />
    </div>
  </Teleport>
</template>

<script lang="ts" setup>
import { ref, inject } from 'vue'
import { useUiStore } from '@core/stores/ui.store'
import { storeToRefs } from 'pinia'

defineProps<{
  disabled?: boolean // To display the slot as-is
  open?: boolean // On mobile, to know if the panel is open
}>()

const emit = defineEmits<{
  close: []
}>()

const uiStore = useUiStore()
</script>

<style lang="postcss" scoped>
.mobile-fullscreen {
  background-color: var(--background-color-secondary);
  position: fixed;
  inset: 0;
  z-index: 100;
}
</style>
```

See `MenuList` below for usage suggestion.